wiki-service/src/test/java/org/exoplatform/wiki/service/TestNoteService.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2003-2010 eXo Platform SAS.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see<http://www.gnu.org/licenses/>.
+ */
+package org.exoplatform.wiki.service;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import org.exoplatform.portal.config.model.PortalConfig;
+import org.exoplatform.services.security.Identity;
+import org.exoplatform.services.security.IdentityConstants;
+import org.exoplatform.wiki.WikiException;
+import org.exoplatform.wiki.jpa.BaseTest;
+import org.exoplatform.wiki.mow.api.*;
+
+public class TestNoteService extends BaseTest {
+  private WikiService wService;
+  private NoteService noteService;
+  public void setUp() throws Exception {
+    super.setUp() ;
+    wService = getContainer().getComponentInstanceOfType(WikiService.class) ;
+    noteService = getContainer().getComponentInstanceOfType(NoteService.class) ;
+    getOrCreateWiki(wService, PortalConfig.PORTAL_TYPE, "classic");
+  }
+  
+
+  public void testGetGroupPageById() throws WikiException {
+    Wiki wiki = getOrCreateWiki(wService, PortalConfig.GROUP_TYPE, "/platform/users");
+    Identity root = new Identity("root");
+
+    assertNotNull(noteService.getNoteOfNoteBookByName(PortalConfig.GROUP_TYPE, "platform/users", "WikiHome")) ;
+
+    try {
+      noteService.createNote(wiki, "WikiHome", new Page("testGetGroupPageById-101", "testGetGroupPageById-101"),root);
+    } catch (IllegalAccessException e) {
+      e.printStackTrace();
```

Comment:
? why not letting the test fail?

Reply:
Because the exception here is related to the note creation, not to the get of the note which is the purpose of the test case, But I think that you are right it will better to fail if the note is not created, I'll update it